test/verify/check-connection

```diff
@@ -1335,11 +1335,15 @@ server {
         m.start_cockpit(tls=True)
         self.checkCockpitOnProxy()
 
+        # don't stop the unit! it tries to shut down gracefully, waiting for the pending cockpit connections
```

Comment:
I ran into a similar problem trying to get aiohttp to shut down in the unit tests, since cockpit.js held the connection to the websocket and the async context manager waited for it to close on exit (which never happened, of course)...

I'd be surprised if nginx didn't have an option to deal with this.  And indeed, the combination of nginx not always exiting in a timely manner and systemd treating this as an unexpected bug (`SIGABRT`) is definitely a bug on the part of whoever wrote the unit file.  We should probably file that somewhere.

Reply:
The unit doesn't fail. The test [failed due to an unexpected message](https://cockpit-logs.us-east-1.linodeobjects.com/pull-20832-ca0a34a0-20240806-210253-fedora-39-other/log.html#113). nginx logs that sigabrt and ABRT catches it. This is sort of an emergent thing, I wouldn't even know where to file it. I lean towards a bug against nginx to but catch SIGABRT and exit cleanly?

Reply:
systemd sends `SIGABRT` under various circumstances related to timeouts or watchdog timeouts because it considers that the process is behaving in a buggy way and wants the core.  The bug is that this is inappropriate for a webserver that's configured to wait indefinitely for websockets to exit.  I'd say the bug should be filed against whoever wrote the unit file (distro or upstream) since that's the point of integration.

Reply:
I filed this as https://bugzilla.redhat.com/show_bug.cgi?id=2304015 . I'll add a reference to the comment / commit.